Automattic/jetpack
# Admin Page: Fix all eslint warnings except the related to using React ES6 classes

Fixes eslint warnings on the Admin Page's code.

After this PR is merged eslint warnings are down from `1193` to `69`. The only remaining warnings are shown all related to the `react/prefer-es6-class` rule.

#### Testing instructions:

1.  Run `node_modules/.bin/eslint --ext .js --ext .jsx _inc/client -c .eslintrc`
2. Confirm that the only warnings you see are related to the `react/prefer-es6-class` rule.
3. Run `yarn build` and confirm that the admin page built ok.

React class warnings are going to be address by #8714.

#### Changes proposed in this Pull Request:

* Fixes spaces and `var` usage  after running `Run eslint --fix`
* Fixes warnings related to the rule `prefer-const`.
* Fixes warnings related to the rule `import-docblock`.
* Fixes warnings related to the rule `brace-style`.
* Fixes warnings related to the rule `operator-linebreak`.
* Fixes warnings related to the rule `brace-style`.
* Makes a line marked to not warn on `react-no-danger`.
* Fixes warnings related to the rule `wpcalypso/i18n-no-this-translate`.
* Fixes warnings related to the rule `valid-jsdoc`.
* Downgrades the `react/no-bind` rule to be ignored temporarily because this pattern is extensively used.
